src/lib/util/feature-evaluator/repository/index.ts

```diff
@@ -38,14 +40,16 @@ export default class Repository {
         bootstrapProvider,
         storageProvider,
     }: RepositoryOptions) {
+        super();
         this.appName = appName;
         this.bootstrapProvider = bootstrapProvider;
         this.storageProvider = storageProvider;
         this.segments = new Map();
     }
 
-    start(): Promise<void> {
-        return this.loadBootstrap();
+    async start(): Promise<void> {
+        await this.loadBootstrap();
+        process.nextTick(() => this.emit(UnleashEvents.Ready));
```

Comment:
A repository needs to emit the ready event to indicate when it has loaded init data.